File: Sources/EchoPluginAPI/NetworkingPlugin/Request.swift

import Foundation

public struct Request: Equatable, Codable, CustomStringConvertible {
    /// Uniquely identifies this request
    public var id: String

    public var url: URL
    public var httpMethod: String
    public var headers: [String: String]

    public var timestamp: Date

    public var humanReadableBody: String
    public var rawBody: Data?

    public var endpoint: Endpoint {
        if #available(iOS 16.0, *) {
            Endpoint(path: url.path(percentEncoded: false))
        } else {
            Endpoint(path: url.path)
        }
    }

    public init(
        id: String,
        url: URL,
        httpMethod: String,
        headers: [String : String],
        timestamp: Date,
        humanReadableBody: String,
        rawBody: Data?
    ) {
        self.id = id
        self.url = url
        self.httpMethod = httpMethod
        self.headers = headers
        self.timestamp = timestamp
        self.humanReadableBody = humanReadableBody
        self.rawBody = rawBody
    }

    public var description: String {
        endpoint.description
    }
}

File: android/plugin-api/src/main/java/xyz/block/echoapp/plugin/networkingv2/internal/Request.kt
package xyz.block.echoapp.plugin.networkingv2.internal

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Request(
    val id: String,
    val timestamp: Long,
    val httpMethod: String,
    val url: String,
    val headers: Map<String, String>,
    val humanReadableBody: String,
    val rawBody: ByteArray? = null,
)
